src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/ProjectModifier.cs

```diff
@@ -0,0 +1,209 @@
+        /// </summary>
+        /// <param name="projectType"></param>
+        /// <returns></returns>
+        public async Task AddAuthCodeAsync()
+        {
+            CodeModifierConfig? codeModifierConfig = GetCodeModifierConfig();
+            //Get CodeModifierConfig from CodeModifierConfigs folder.
+            if (codeModifierConfig != null &&
+                codeModifierConfig.Files != null &&
+                !string.IsNullOrEmpty(_toolOptions.ProjectFilePath))
+            {
+                //Initialize CodeAnalysis.Project wrapper
+                CodeAnaylsisHelper project = new CodeAnaylsisHelper(_toolOptions.ProjectFilePath);
+
+                //Go through all the files, make changes using DocumentBuilder.
+                foreach(var file in codeModifierConfig.Files)
+                {
+                    var fileName = file.FileName;
+                    string className = GetClassName(fileName);
+
+                    //if the file we are modifying is Startup.cs, use Program.cs to find the correct file to edit.
+                    if (!string.IsNullOrEmpty(file.FileName) && file.FileName.Equals("Startup.cs", StringComparison.OrdinalIgnoreCase))
+                    {
+                        fileName = await GetStarupClass(project.CodeAnalysisProject);
+                    }
+
+                    if (!string.IsNullOrEmpty(fileName))
+                    {
+                        string? filePath = Directory.EnumerateFiles(_toolOptions.ProjectPath, fileName, SearchOption.AllDirectories).FirstOrDefault();
+                        //get the file document to get the document root for editing.
+                        var fileDoc = project.CodeAnalysisProject.Documents.Where(d => d.Name.Equals(filePath)).FirstOrDefault();
+                        if (!string.IsNullOrEmpty(filePath) && fileDoc != null)
+                        {
+                            DocumentEditor documentEditor = await DocumentEditor.CreateAsync(fileDoc);
+                            DocumentBuilder documentBuilder = new DocumentBuilder(documentEditor, file, _consoleLogger);
+                            var docRoot = documentEditor.OriginalRoot as CompilationUnitSyntax;
+                            var namespaceNode = docRoot?.Members.OfType<NamespaceDeclarationSyntax>()?.FirstOrDefault();
+
+                            //get classNode. All class changes are done on the ClassDeclarationSyntax and then that node is replaced using documentEditor.
+                            var classNode = namespaceNode?
+                                .DescendantNodes()?
+                                .Where(node =>
+                                    node is ClassDeclarationSyntax cds &&
+                                    cds.Identifier.ValueText.Contains(className))
+                                .FirstOrDefault();
+
+                            if (classNode != null && classNode is ClassDeclarationSyntax classDeclarationSyntax)
```

Comment:
>classNode != null  [](http://example.com/codeflow?start=32&length=18)

not needed

Reply:
fixed and moved https://github.com/dotnet/Scaffolding/pull/1593

Reply:
fixed and moved https://github.com/dotnet/Scaffolding/pull/1593

Reply:
fixed and moved to https://github.com/dotnet/Scaffolding/pull/1593/